core/src/main/java/com/linecorp/armeria/internal/client/AbstractClientOptionsBuilder.java

```diff
@@ -380,7 +386,7 @@ protected final ClientOptions buildOptions(boolean includeDefault) {
         if (includeDefault) {
             return ClientOptions.of(optValArray);
         } else {
-            return ClientOptions.of(ClientOptions.EMPTY, optValArray);
+            return ClientOptions.of(ClientOptions.empty(), optValArray);
```

Comment:
Hmm, I'm not sure if it's a good idea to expose `empty()` in the public API. How about this:

- Remove `EMPTY` and `empty()`
- Change the signiture of `buildOptions()` to:
  - `buildOptions(@Nullable ClientOptions baseOptions)` (We pass `WebClient.options()` in this PR, or `null` for other cases.)
- Use `ClientOptions.of(baseOptions, optValArray)` (if `null`, use `ClientOptions.of(optValArray)`.

Reply:
> I'm not sure if it's a good idea to expose empty()

That was my considering point. `buildOptions(@Nullable ClientOptions baseOptions)` sounds good.
Thanks for reviewing 👍